### PR TITLE
Fix flickering popup menu

### PR DESCRIPTION
**Pull request by tamy0612**
## Problem
The popup menu is blinking/flickering often.

## Cause
- there is no reset code for `s:has_poped_up`
- `s:remote_refresh` calls `s:core_complete` via `s:python_refresh_completions` when the flag `g:asyncomplete_smart_completion` is set
- Related to above, `s:python_refresh_completions` is not called even if the flag `'refresh'` is set.

## Solution
- `l:has_poped_up` -> `s:has_poped_up`, since `l:has_poped_up` is never used and it seems a typo
- Modify the condition to call `s:python_refresh_completions` in `s:remote_refresh`

## Comment
The solution is just my guess.  Although it is working on my PC, I could have misunderstanding on the design of this plugin.

### CODE REVIEW

File: autoload/asyncomplete.vim [hunk 1]
```diff
@@ -278,9 +278,8 @@ function! s:remote_refresh(ctx, force) abort
         let l:typed_len = l:endpos - l:startpos
         if l:typed_len == 1
             call add(l:sources_to_notify, l:name)
-        elseif has_key(s:matches, l:name) && s:matches[l:name]['refresh']
-            call add(l:sources_to_notify, l:name)
-        elseif s:supports_smart_completion()
+        endif
+        if has_key(s:matches, l:name) && s:matches[l:name]['refresh']
```
Comment (prabirshrestha) [inline autoload/asyncomplete.vim]: This one refresh all the sources. We should only refresh the source that has `refresh=1`. can you try with multiple sources and check the logs.

File: autoload/asyncomplete.vim [hunk 2]
```diff
@@ -278,9 +278,8 @@ function! s:remote_refresh(ctx, force) abort
         let l:typed_len = l:endpos - l:startpos
         if l:typed_len == 1
             call add(l:sources_to_notify, l:name)
-        elseif has_key(s:matches, l:name) && s:matches[l:name]['refresh']
-            call add(l:sources_to_notify, l:name)
-        elseif s:supports_smart_completion()
```
Comment (prabirshrestha) [inline autoload/asyncomplete.vim]: `smart completion` is required when every keystroke needs to refresh the list. hence we do need it. You need lua.

File: autoload/asyncomplete.vim (hunk 1 above)
Comment (y7amura) [reply autoload/asyncomplete.vim]: I think this is because of `if l:typed_len == 1`.  What is the purpose of this `if` block?

**Comment by prabirshrestha** [comment]
Here is briefly at how it works. I have been planning to re-architect asyncomplete library to be more efficient and also allows write the core engine in pure vimscript and in other language so that things like fuzzy search can be done in a different process.

asyncomplete

**Comment by tamy0612** [comment]
Thank you for your kind explanation, and sorry that I put comments with another account.  I've just removed the 2nd commit which is the problem discussed in this thread.

**Pull request merged**